packages/cli/src/util/dev/server.ts

```diff
@@ -675,31 +675,48 @@ export default class DevServer {
       this.getLocalEnv('.env.build', vercelConfig.build?.env),
     ]);
 
-    let allEnv = { ...buildEnv, ...runEnv };
-
     // If no .env/.build.env is present, use cloud environment variables
-    if (Object.keys(allEnv).length === 0) {
+    if (Object.keys({ ...runEnv, ...buildEnv }).length === 0) {
```

Comment:
```suggestion
    const keyLength = Object.keys(runEnv).length + Object.keys(buildEnv).length
    if (keyLength === 0) {
```

Small change to avoid iterating more times than necessary